docs/source/traits_user_manual/defining.rst

```diff
@@ -663,12 +663,18 @@ HasTraits objects:
   for more information on trait editors.
 * **label**: A string providing a human-readable name for the trait. It is
   used to label trait attribute values in user interface trait editors.
-* **rich_compare**: A Boolean indicating whether the basis for considering a
-  trait attribute value to have changed is a "rich" comparison (True, the
-  default), or simple object identity (False). This attribute can be useful
-  in cases where a detailed comparison of two objects is very expensive, or
-  where you do not care if the details of an object change, as long as the
-  same object is used.
+* **comparison_mode**: Indicates when trait change notifications should be
+  generated based upon the result of comparing the old and new values of a
+  trait assignment:
+
+  * 0 (NO_COMPARE): The values are not compared and a trait change
+    notification is generated on each assignment.
+  * 1 (OBJECT_IDENTITY_COMPARE): A trait change notification is
+    generated if the old and new values are not the same object.
+  * 2 (RICH_COMPARE): A trait change notification is generated if the
+    old and new values are not equal using Python's
+    'rich comparison' operator. This is the default.
```

Comment:
I'd just leave this as "are not equal." and lose the rest of the sentence. Python 3 only has rich comparisons, so there's only one obvious way to test for equality.

Reply:
This also now needs updating for the new enumerations.

Reply:
I've updated this to point to the `ComparisonMode` documentation, and avoided repeating the meaning of the values here.